airflow/models/baseoperator.py

```diff
@@ -670,6 +672,7 @@ def __deepcopy__(self, memo):
 
         for k, v in self.__dict__.items():
             if k not in shallow_copy:
+                # noinspection PyArgumentList
```

Comment:
Do we need this? I don't think we need this check. We can disable this check on our Pycharm/Intelij

Reply:
I am afraid so. The check is super useful because it detects spelling mistakes in parameters or removed parameters. 

This is actually a known problem because the memo parameter is by mistake missing in skeleton definition for python skeletons (from typeshed I believe). The issue is described
here: https://intellij-support.jetbrains.com/hc/en-us/community/posts/360000181224-deepcopy-unexpected-memo-argument
and reported here (unresolved):  https://youtrack.jetbrains.com/issue/PY-29664 

I have not found a good way of silencing it only for deepcopy, so I propose we can leave it as it is.